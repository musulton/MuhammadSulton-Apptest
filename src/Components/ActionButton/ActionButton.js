import * as React from 'react';
import {Text, TouchableOpacity, View} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

import styles from './ActionButton.styles';
import config from './ActionButton.config';
import {Separator} from '../index';

const renderIcon = type => (
  <View style={styles.iconContainer}>
    <Icon name={config.PROPERTY_ACTION[type].icon} size={20} color="#fff" />
  </View>
);

const renderText = type => (
  <View style={styles.textContainer}>
    <Text>{config.PROPERTY_ACTION[type].title}</Text>
    <Text>{config.PROPERTY_ACTION[type].description}</Text>
    {/*<View style={styles.distance}>*/}
    {/*  <Separator />*/}
    {/*</View>*/}
  </View>
);

const ActionButton = ({type, onPress}) => (
  <TouchableOpacity style={styles.container} onPress={onPress}>
    {renderIcon(type)}
    {renderText(type)}
  </TouchableOpacity>
);

export default ActionButton;
