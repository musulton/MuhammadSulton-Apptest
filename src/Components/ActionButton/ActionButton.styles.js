import {StyleSheet} from 'react-native';

import Colors from '../../Themes/Colors';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    marginVertical: 10,
  },
  iconContainer: {
    width: 40,
    height: 40,
    backgroundColor: Colors.actionIcon,
    borderRadius: 500,
    justifyContent: 'center',
    alignItems: 'center',
  },
  textContainer: {
    marginLeft: 10,
  },
  text: {
    color: Colors.text,
  },
});
