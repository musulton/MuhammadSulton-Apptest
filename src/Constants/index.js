export default {
  ROUTES: {
    FORM: 'FORM',
    DETAILS: 'Details',
    LIST: 'List',
  },
  ACTION_BUTTON: {
    ADD: 'ADD',
    EDIT: 'EDIT',
    DELETE: 'DELETE',
  },
  REDUX_ACTION: {
    SET_CONTACT: 'SET_CONTACT',
  },
};
