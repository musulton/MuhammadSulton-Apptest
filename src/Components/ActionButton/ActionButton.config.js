import Constants from '../../Constants';

const PROPERTY_ACTION = {
  [Constants.ACTION_BUTTON.ADD]: {
    icon: 'plus',
    title: 'Add Contact',
    description: 'More contact the better',
  },
  [Constants.ACTION_BUTTON.EDIT]: {
    icon: 'pencil',
    title: 'Edit Contact',
    description: 'You can change everything',
  },
  [Constants.ACTION_BUTTON.DELETE]: {
    icon: 'trash',
    title: 'Delete Contact',
    description: 'Too bad to cut off communication',
  },
};

export default {
  PROPERTY_ACTION,
};
