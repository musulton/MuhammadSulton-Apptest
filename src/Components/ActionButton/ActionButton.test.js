import * as React from 'react';
import {render, cleanup} from '@testing-library/react-native';

import Constants from '../../Constants';

import ActionButton from './ActionButton';

describe('ActionButton', () => {
  afterEach(cleanup);

  it('should render add type correctly', () => {
    const {toJSON} = render(
      <ActionButton type={Constants.ACTION_BUTTON.ADD} onPress={jest.fn} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render edit type correctly', () => {
    const {toJSON} = render(
      <ActionButton type={Constants.ACTION_BUTTON.EDIT} onPress={jest.fn} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render delete type correctly', () => {
    const {toJSON} = render(
      <ActionButton type={Constants.ACTION_BUTTON.DELETE} onPress={jest.fn} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
