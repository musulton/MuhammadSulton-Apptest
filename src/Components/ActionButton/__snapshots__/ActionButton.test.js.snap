// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton should render add type correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginVertical": 10,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#547980",
        "borderRadius": 500,
        "height": 40,
        "justifyContent": "center",
        "width": 40,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "marginLeft": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
        }
      }
    >
      Add Contact
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
        }
      }
    >
      More contact the better
    </Text>
  </View>
</View>
`;

exports[`ActionButton should render delete type correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginVertical": 10,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#547980",
        "borderRadius": 500,
        "height": 40,
        "justifyContent": "center",
        "width": 40,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "marginLeft": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
        }
      }
    >
      Delete Contact
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
        }
      }
    >
      Too bad to cut off communication
    </Text>
  </View>
</View>
`;

exports[`ActionButton should render edit type correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginVertical": 10,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#547980",
        "borderRadius": 500,
        "height": 40,
        "justifyContent": "center",
        "width": 40,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "marginLeft": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
        }
      }
    >
      Edit Contact
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
        }
      }
    >
      You can change everything
    </Text>
  </View>
</View>
`;
